src/lib.rs

```diff
@@ -409,6 +463,26 @@ impl Core {
     }
 }
 
+#[inline]
+fn check_has_reader(state: usize) -> bool {
+    state & HAS_READER == HAS_READER
+}
+
+#[inline]
+fn set_has_reader(state: usize) -> usize {
+    state | HAS_READER
+}
+
+#[inline]
+fn clear_has_reader(state: usize) -> usize {
+    state & !HAS_READER
+}
+
+#[inline]
+fn wrapping_add(a: usize, b: usize) -> usize {
+    (a + b) & MAX_CAPACITY
+}
```

Comment:
There are now enough operations on `state` that I wonder if it's worth turning it into a newtype...but, we can do that in a separate PR.